test/integration/targets/podman_container/tasks/main.yml

```
@@ -0,0 +1,288 @@
+          - deleted_again.ansible_facts.podman_container == {}
+          - deleted_again.actions == []
+        fail_msg: "Deleting started container test failed!"
+        success_msg: "Deleting started container test passed!"
+
+    - name: Recreate container with parameters
+      podman_container:
+        name: container
+        image: alpine:3.7
+        state: started
+        command: sleep 1d
+        recreate: true
+        etc_hosts:
+          host1: 127.0.0.1
+          host2: 127.0.0.1
+        annotation:
+          this: "annotation_value"
+        dns:
+          - 1.1.1.1
+          - 8.8.4.4
+        dns_search: example.com
+        cap_add:
+          - SYS_TIME
+          - NET_ADMIN
+        publish:
+          - "9000:80"
+          - "9001:8000"
+        workdir: "/bin"
+        env:
+          FOO: bar
+          BAR: foo
+        group_add: "somegroup"
+        label:
+          somelabel: labelvalue
+          otheralbe: othervalue
+        volumes:
+          - /tmp:/data
+      register: test
+
+    - name: Check output is correct
+      assert:
+        that:
+          - test is changed
+          - test.ansible_facts is defined
+          - test.ansible_facts.podman_container is defined
+          - test.ansible_facts.podman_container != {}
+          - test.ansible_facts.podman_container['State']['Running'] == true
+          # test capabilities
+          - "'CAP_SYS_TIME' in test.ansible_facts.podman_container['BoundingCaps']"
+          - "'CAP_NET_ADMIN' in test.ansible_facts.podman_container['BoundingCaps']"
+          # test annotations
+          - test.ansible_facts.podman_container['Config']['Annotations']['this'] is defined
+          - test.ansible_facts.podman_container['Config']['Annotations']['this'] == "annotation_value"
+          # test DNS
+          - test.ansible_facts.podman_container['HostConfig']['DNS'] is defined
+          - test.ansible_facts.podman_container['HostConfig']['DNS'] == ['1.1.1.1', '8.8.4.4']
+          # test ports
+          - test.ansible_facts.podman_container['NetworkSettings']['Ports']|length == 2
+          # test working dir
+          - test.ansible_facts.podman_container['Config']['WorkingDir'] == "/bin"
+          # test dns search
+          - test.ansible_facts.podman_container['HostConfig']['DNSSearch'] == ['example.com']
+          # test environment variables
+          - "'FOO=bar' in test.ansible_facts.podman_container['Config']['Env']"
+          - "'BAR=foo' in test.ansible_facts.podman_container['Config']['Env']"
+          # test labels
+          - test.ansible_facts.podman_container['Config']['Labels'] | length == 2
+          - test.ansible_facts.podman_container['Config']['Labels']['somelabel'] == "labelvalue"
+          - test.ansible_facts.podman_container['Config']['Labels']['otheralbe'] == "othervalue"
+          # test mounts
+          - "'/data' in test.ansible_facts.podman_container['Mounts'] | map(attribute='destination') | list"
```

Pro tip: to avoid this double-single quotes mixture and improve readability I tend to use a YAML block like this:
```suggestion
          - >-
            '/data' in test.ansible_facts.podman_container['Mounts'] | map(attribute='destination') | list
```
this allows to drop outer quotes and is simpler in general when it comes to escaping weird things.

---

thanks, really need this now